### PR TITLE
Update rush version, upgrade Node.js GA action version, remove Node 12 (closes #1093)

### LINKED ISSUE #1093: Updates/Upgrades in tooling and CI (Node.js, rush)
**Is your feature request related to a problem? Please describe.**
Update tooling versions to benefit from enhancements.

**Describe the solution you'd like**
Update:
- Rush version to be able to use [rush-pnpm](https://github.com/microsoft/rushstack/issues/2347#issuecomment-1152800580).
- Version requirements for using the repo through rush.json to allow Node.js 14 & 16.

Upgrade:
- Node.js version on GH Actions pipeline.

Remove:
- Support for Node.js v12 in the repository.

**Additional context**
- `rush-pnpm` will make it more easier and natively supported to run maintainer type commands such as `rush-pnpm audit`, `rush-pnpm outdated` etc.
- Improve build & test times:

| Command | Node v14 Avg. Time (3 runs) | Node v16 Avg. Time (3 runs) | 
| --- | --- | --- |
| `rush build` | ~64s | ~38s |
| `rush test` | ~15s | ~10s |

### PR DESCRIPTION
**Pull request by igneel64**
## Changes
- Remove Node 12 repository support.
- Update Rush version to have `rush-pnpm` available.
- Upgrade Node 16 for CI builds.
- Added .nvmrc to make it less of a pain for maintainers.

Closes #1093 (See for more info)

### Note for existing maintainers
Run:
- `rush version` 
- Re-run `npm install --global @microsoft/rush`
- try `rush-pnpm audit` or some other command

### CODE REVIEW

**Comment by snowplowcla** [comment]
Thanks for your pull request. Is this your first contribution to a Snowplow open source project? Before we can look at your pull request, you'll need to sign a Contributor License Agreement (CLA).

:memo: Please visit https://docs.snowplowanalytics.com/docs/contributing/contributor-license-agreement/ to learn more and sign.

Once you've signed, please reply here (e.g. I signed it!) and we'll verify. Thanks.

File: README.md [hunk 1]
```diff
@@ -51,7 +51,7 @@ client-side and server-side tiers of your websites and web apps.
 
 ### Maintainer quick start
 
-Assuming [git](https://git-scm.com/downloads), [Node.js 12 LTS or 14 LTS](https://nodejs.org/en/download/releases/) are installed.
+Assuming [git](https://git-scm.com/downloads), [Node.js14 or 16](https://nodejs.org/en/download/releases/) are installed.
```
Comment (paulboocock) [inline README.md:54]:
```suggestion
Assuming [git](https://git-scm.com/downloads), [Node.js 14 or 16](https://nodejs.org/en/download/releases/) are installed.
```

**Review by paulboocock — APPROVED** [review]

**Comment by paulboocock** [comment]
If I'm being nitpicky, I'd perhaps shorten this commit name a little but that might just be me. "Upgrade Node.js and rush versions" is probably descriptive enough.

**Review by matus-tomlein — APPROVED** [review]
LGTM!

**Pull request merged**